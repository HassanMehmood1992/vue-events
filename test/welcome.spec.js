import { mount, createLocalVue } from "@vue/test-utils";
import { shallowMount } from "@vue/test-utils";
import Vuetify from "vuetify";
import Vuex from "vuex";

import Welcome from "@/components/welcome";

const localVue = createLocalVue();
localVue.use(Vuex);

let wrapper;

beforeEach(() => {
  let vuetify = new Vuetify();

  wrapper = shallowMount(Welcome, {
    store: new Vuex.Store({
      state: { products: [] }
    }),
    localVue,
    vuetify,
    stubs: {
      NuxtLink: true,
      Nuxt: true
      // Any other component that you want stubbed
    }
  });
});

afterEach(() => {
  wrapper.destroy();
});

describe("Welcome component", () => {
  test("is fully functional", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test("should have a #RoadSurferAssignment", () => {
    const title = wrapper.find(".pa-4.ma-3");
    expect(title.element.innerHTML).toBe("#RoadSurferAssignment");
  });
});
