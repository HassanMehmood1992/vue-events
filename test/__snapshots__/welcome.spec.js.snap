// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component is fully functional 1`] = `
<v-container
  class="ma-0 pa-0"
  fluid=""
>
  <v-img
    lazy-src="/images/roadsurfer-campervan-hire.jpg"
    min-height="50vh"
    src="/images/roadsurfer-campervan-hire.jpg"
  >
    <v-layout
      class="my-10"
      justify-end=""
      row=""
    >
      <v-flex
        class="secondary pa-4 ma-3 title white--text"
        md3=""
      >
        #RoadSurferAssignment
      </v-flex>
    </v-layout>
  </v-img>
</v-container>
`;
